### ISSUE TITLE
Add a Call menu

### ISSUE THREAD

**Issue by pixlwave** [jitsi/jitsi-meet-electron#692]
## Description
I'd love to see a call menu added to the app with options to mute/toggle camera/leave etc. This would make the app feel more at home on macOS where common commands are expected to be found in the menubar, along with information about any keyboard shortcuts that apply to them.

I would be more than happy to start the implementation off for this with a Leave menu item if someone could point me to what should be called from `main.js` to leave a call as I couldn't figure this part out.

```javascript
        ...
        {
            label: 'Call',
            submenu: [ {
                label: 'Leave',
                accelerator: 'CmdOrCtrl+L',
                click: function(menuItem, browserWindow, event) {
                    // tell the browser window to leave the call?
                }
            } ]
        },
        ...
```

## Current behaviour
The common commands are only available in the floating toolbar that disappears when the mouse is outside of the window.

## Possible Solution
![Screenshot 2021-12-27 at 1 16 15 pm](https://user-images.githubusercontent.com/6060466/147475799-cb2c114a-c48a-451d-90f3-219697b8f39d.png)

# Environment details
Jitsi 2021.12.2, macOS 11

**Comment by saghul** [comment]
Not sure what this adds to be honest.

If you need to search for the app to focus it before going to its menu, then you are already there so you can take the primary action, from within the app itself. Or alternatively use the always on top window.

A dedicated menu bar icon with action would be more useful IMHO, because it's available regardless of the focused application.

**Comment by pixlwave** [comment]
Thanks for the reply @saghul.

I think the section about app-specific menus in the [HIG](https://developer.apple.com/design/human-interface-guidelines/macos/menus/menu-bar-menus/) probably outlines why this is a good idea better than I ever could:

> **Provide app-specific menus for custom commands.** People look in the menu bar when searching for app-specific commands, especially when using an app for the first time. Even when commands are available elsewhere in your app, it’s important to list them in the menu bar. Putting commands in the menu bar makes them easier for people to find, lets you assign keyboard shortcuts to them, and makes them more accessible to people using Full Keyboard Access. Excluding commands from the menu bar — even infrequently used or advanced commands — risks making them difficult for everyone to find.

From my perspective, in addition to creating a familiar experience on macOS, it would mean users could map their own keyboard shortcuts in System Preferences | Keyboard | Shortcuts as well as add basic automations to the app using AppleScript.

I'm not sure whether you already have VoiceOver and Voice Control support covered, but it's worth noting that menu bar items are exposed to both of these for free too.

Thanks again :)

**Comment by pixlwave** [comment]
Oh and on this:

> A dedicated menu bar icon with action would be more useful IMHO, because it's available regardless of the focused application.

I agree that would also be a lovely addition for people who are using the app regularly. It might be interesting to consider a couple of custom items for the dock context menu too?

**Comment by u35253** [comment]
> A dedicated menu bar icon with action would be more useful IMHO, because it's available regardless of the focused application.

Yes, a menu bar icon that shows the status ("green circle = Online", "gray circle = Offline", etc.) would go a very long way for me.  This would allow glancing to see whether Jitsi is running and thinks it is online.  (I always hide the dock fully for space, and I use the menu bar extensively for things like this.)

"Actions" from a menu bar icon would be nice-to-haves or "next steps".  Just "seeing it" in the menu bar would be huge for me.  But, actions for a menu from the menu bar icon that come to mind off the top of my head might be:

- Show Jitsi
- Set Status
- Dial Pad [Show Jitsi and open Dial Pad]
- Call History [Show Jitsi and open Call History]

I don't know how to implement this myself, but I prefer Jitsi over all the other macOS VOIP apps, so this would be amazing.
